Cosmic Shiv Rework (from the person the item is dedicated to)

YataYT opened the pull request:
Rework from Yatagarasu (patreon donor for Cosmic Shiv), which now shoots a homing star trail to a target, spawning an invisible, non-damaging aura that sticks to the target, continously spawning extremely fast blades for 5 seconds. I didn't add a limit for how many auras can exist at once, but I added limits to how many can be applied at once. Galaxy Star is no longer a thing, but the texture is used for Cosmic Shiv Trail now, so I renamed the file to match. Not multiplayer-tested since I have no friends and my PC is too beefy to know how laggy it might be for weaker systems.

My intention with this weapon is to be a slow firing projectile-focused weapon that can stack up a bit on one target, and be able to keep up with Yharon, hence the accelerating homing and high speed blades. Feel free to nerf the penetration of the blades if they're too op against Thanatos.

This one was a big step-up from my previous projects (which was basically just making very basic weapons or slightly modifying ExampleMod code, and I'm overall extremely happy with how it went. Code might be pretty bad since I only had 2 weeks of modding experience by the time I finished this, so the code might need some cleaning up, and if the draws are too laggy, feel free to nerf the values a bit (although I tweaked with them a lot to get them looking the best)

YataYT committed:
Rework from Yatagarasu (patreon donor for Cosmic Shiv), which now shoots a homing star trail to a target, spawning an invisible, non-damaging aura that sticks to the target, continously spawning extremely fast blades for 5 seconds. I didn't add a limit for how many auras can exist at once, but I added limits to how many can be applied at once. Galaxy Star is no longer a thing, but the texture is used for Cosmic Shiv Trail now, so I renamed the file to match.

My intention with this weapon is to be a slow firing projectile-focused weapon that can stack up a bit on one target, and be able to keep up with Yharon, hence the accelerating homing and high speed blades. Feel free to nerf the penetration of the blades if they're too op against Thanatos.

This one was a big step-up from my previous projects (which was basically just making very basic weapons or slightly modifying ExampleMod code, and I'm overall extremely happy with how it went. Code might be pretty bad since I only had 2 weeks of modding experience by the time I finished this, so the code might need some cleaning up, and if the draws are too laggy, feel free to nerf the values a bit (although I tweaked with them a lot to get them looking the best)

YataYT committed:
Fixed Cosmic Shiv trail homing

After further testing, I realized the trail freezes when its velocity gets low enough, due to having to change directions to reach its target

Ozzatron commented on Projectiles/Melee/Shortswords/CosmicShivProj.cs:
```diff
@@ -50,31 +57,65 @@ public override void SetVisualOffsets()
-                Main.dust[dustID].scale *= 2f;
-            }
-            target.AddBuff(ModContent.BuffType<GodSlayerInferno>(), 120);
+            // Congrats on trying to PvP with this
+            target.AddBuff(ModContent.BuffType<GodSlayerInferno>(), 216000);
+            target.AddBuff(ModContent.BuffType<VulnerabilityHex>(), 216000);
+            target.AddBuff(ModContent.BuffType<WhisperingDeath>(), 216000);
+            target.AddBuff(ModContent.BuffType<CrushDepth>(), 216000);
+            target.AddBuff(ModContent.BuffType<Nightwither>(), 216000);
+            target.AddBuff(ModContent.BuffType<VulnerabilityHex>(), 216000);
+            target.AddBuff(ModContent.BuffType<GlacialState>(), 600);           // Not that evil
+            target.AddBuff(BuffID.Stoned, 600);                                 // But just in case you're immune to glacial state
```
Remove this function.

Ozzatron reviewed (changes requested):
Remove the PvP meme function and it looks good.

YataYT committed:
okay :c

YataYT committed:
Revised CosmicShivAura

Added .cs prefix to file, added a null check to !target.active, and also kills the projectile when the target can no longer take damage

YataYT committed:
Never mind, keep trying even if target is invincible

Ozzatron commented on Projectiles/Melee/CosmicShivAura.cs:
```diff
@@ -44,7 +44,7 @@ public override void OnSpawn(IEntitySource source)
         {
             // Unexist with target
-            if (!target.active || target is null) {
+            if (!target?.active == true || target is null) {
```
By flipping these cases around you don't need to guard against target being null twice.
```suggestion
            if (target is null || !target.active) {
```

Ozzatron reviewed (dismissed):
Looks good except for that if statement, which I've got handled.

Ozzatron committed:
Flip target guard if statement
